app/src/main/java/org/wikipedia/views/SurveyDialog.kt

```diff
@@ -0,0 +1,117 @@
+
+    fun showFeedbackOptionsDialog(activity: Activity, source: Constants.InvokeSource) {
+        sendAnalyticsEvent("toolbar_first_feedback", "pt_feedback", source)
+        var dialog: AlertDialog? = null
+        val feedbackView = activity.layoutInflater.inflate(R.layout.dialog_patrol_edit_feedback_options, null)
+
+        val clickListener = View.OnClickListener {
+            val feedbackOption = (it as TextView).text.toString()
+            dialog?.dismiss()
+            if (
+                (source == Constants.InvokeSource.PLACES &&
+                        feedbackOption != activity.getString(R.string.patroller_diff_feedback_dialog_option_unsatisfied) &&
+                        feedbackOption != activity.getString(R.string.places_survey_very_unsatisfied_option)) ||
+                (source != Constants.InvokeSource.PLACES && feedbackOption == activity.getString(R.string.patroller_diff_feedback_dialog_option_satisfied))
+            ) {
+                showFeedbackSnackbarAndTooltip(activity, source)
+            } else {
+                showFeedbackInputDialog(activity, source, feedbackOption)
+            }
+
+            sendAnalyticsEvent("feedback_selection", "pt_feedback", source,
+                PatrollerExperienceEvent.getActionDataString(feedbackOption = feedbackOption))
+        }
+
+        feedbackView.findViewById<TextView>(R.id.optionSatisfied).setOnClickListener(clickListener)
+        feedbackView.findViewById<TextView>(R.id.optionNeutral).setOnClickListener(clickListener)
+        feedbackView.findViewById<TextView>(R.id.optionUnsatisfied).setOnClickListener(clickListener)
+        if (source == Constants.InvokeSource.PLACES) {
+            val verySatisfied = feedbackView.findViewById<TextView>(R.id.optionVerySatisfied)
+            val veryUnsatisfied = feedbackView.findViewById<TextView>(R.id.optionVeryUnsatisfied)
+            verySatisfied.visibility = View.VISIBLE
+            veryUnsatisfied.visibility = View.VISIBLE
+            verySatisfied.setOnClickListener(clickListener)
+            veryUnsatisfied.setOnClickListener(clickListener)
+        }
+        PatrollerExperienceEvent.logImpression("pt_feedback")
+        val dialogBuilder = MaterialAlertDialogBuilder(activity)
+            .setTitle(
+                if (source == Constants.InvokeSource.PLACES) R.string.places_survey_dialog_title else
+                    R.string.patroller_diff_feedback_dialog_title
+            )
+            .setCancelable(false)
+            .setView(feedbackView)
+        if (source == Constants.InvokeSource.PLACES) {
+            dialogBuilder.setNegativeButton(R.string.logged_out_in_background_cancel) { _, _ ->
+                dialog?.dismiss()
+                sendAnalyticsEvent("feedback_cancel", "pt_feedback", source)
+            }
+        }
+        dialog = dialogBuilder.show()
+    }
+
+    private fun showFeedbackInputDialog(activity: Activity, source: Constants.InvokeSource, feedbackOption: String) {
```

Comment:
The `feedbackOption` is not used. Have you checked if this change affects the Patroller Tasks survey?